string.c

```diff
@@ -8022,37 +8022,50 @@ rb_str_tr_bang(VALUE str, VALUE src, VALUE repl)
+ *  translated to the corresponding character in string +replacements+.
+ *  The correspondence is _positional_:
+ *
+ *  - Each occurrence of the first character in +replaceables+
+ *    is translated to the first character in +replacements+.
+ *  - Each occurrence of the second character in +replaceables+
+ *    is translated to the second character in +replacements+.
+ *  - And so on.
+ *
+ *  Example:
+ *
+ *    'hello'.tr('el', 'ip') #=> "hippo"
+ *
+ *  If +replacements+ is shorter than +replaceables+,
+ *  it is implicitly padded with its own last character:
+ *
+ *    'hello'.tr('aeiou', '-')   # => "h-ll-"
+ *    'hello'.tr('aeiou', 'AA-') # => "hAll-"
  *
- *  Returns a copy of +str+ with the characters in +from_str+ replaced by the
- *  corresponding characters in +to_str+.  If +to_str+ is shorter than
- *  +from_str+, it is padded with its last character in order to maintain the
- *  correspondence.
+ *  Three characters may get special treatment:
  *
- *     "hello".tr('el', 'ip')      #=> "hippo"
- *     "hello".tr('aeiou', '*')    #=> "h*ll*"
- *     "hello".tr('aeiou', 'AA*')  #=> "hAll*"
+ *  - A circumflex (<tt>'^'</tt>) in +replaceables+
+ *    operates as a "not" operator for the following characters in +replaceables+:
  *
- *  Both strings may use the <code>c1-c2</code> notation to denote ranges of
- *  characters, and +from_str+ may start with a <code>^</code>, which denotes
- *  all characters except those listed.
+ *      'hello'.tr('^aeiou', '-') # => "-e--o"
+ *      'hello'.tr('ae^iou', '-') # => "h-ll-"
  *
- *     "hello".tr('a-y', 'b-z')    #=> "ifmmp"
- *     "hello".tr('^aeiou', '*')   #=> "*e**o"
+ *  - A hyphen (<tt>'-'</tt>) embedded in a 3-character +replaceables+ or +replacements+
+ *    defines a range of characters instead of a plain string of characters:
  *
- *  The backslash character <code>\\</code> can be used to escape
- *  <code>^</code> or <code>-</code> and is otherwise ignored unless it
- *  appears at the end of a range or the end of the +from_str+ or +to_str+:
+ *      'ibm'.tr('b-z', 'a-z') # => "hal"
  *
- *     "hello^world".tr("\\^aeiou", "*") #=> "h*ll**w*rld"
- *     "hello-world".tr("a\\-eo", "*")   #=> "h*ll**w*rld"
+ *  - A backslash (<tt>'\'</tt>) before a circumflex, a hyphen, or another backslash
+ *    operates as an escape character:
  *
- *     "hello\r\nworld".tr("\r", "")   #=> "hello\nworld"
- *     "hello\r\nworld".tr("\\r", "")  #=> "hello\r\nwold"
- *     "hello\r\nworld".tr("\\\r", "") #=> "hello\nworld"
+ *      'hel^lo'.tr('\^aeiou', '-')  # => "h-l-l-"    # Escaped leading circumflex.
+ *      'i-b-m'.tr('b\-z', 'a-z')    # => "ibabm"     # Escaped embedded hyphen.
+ *      'foo\\bar'.tr('ab\\', 'XYZ') # => "fooZYXr"   # Escaped backslash.
+ *      'abc\ndef'.tr('a-z', 'b-z')  # => "bcd\\oefg" # '\n' is two characters.
+ *      "abc\ndef".tr('a-z', 'b-z')  # => "bcd\nefg"  # "\n" is one newlne character.
```

Comment:
I'm not sure what these two examples are trying to show except that single quoted strings treat backslash characters literally whereas double quotes treat them as a control character.

Maybe the original examples showing how control characters can be replaced should be kept.

```
 *     "hello\r\nworld".tr("\r", "")   #=> "hello\nworld"
 *     "hello\r\nworld".tr("\\r", "")  #=> "hello\r\nwold"
 *     "hello\r\nworld".tr("\\\r", "") #=> "hello\nworld"
```